_posts/2020-07-12-ray-tune.md

```diff
@@ -0,0 +1,321 @@
+[auto-sklearn], both of which offer advanced hyperparameter optimization
+features. Other implementations that don't follow the Scikit-learn interface
+include include [Ray Tune], [AutoML] and [Optuna].
+
+[AutoML]:https://www.automl.org/
+[auto-sklearn]:https://automl.github.io/auto-sklearn/master/
+
+[Ray] recently provided a wrapper to [Ray Tune] that mirrors the Scikit-learn
+API. [The introduction][1] of this library states the following:
+
+[Ray]:https://docs.ray.io
+[Ray Tune]:https://docs.ray.io/en/master/tune.html
+
+[dml-ms]:https://ml.dask.org/hyper-parameter-search.html
+[skl-ms]:https://scikit-learn.org/stable/modules/grid_search.html
+[tsne]:https://distill.pub/2016/misread-tsne/
+[tuning]:https://scikit-learn.org/stable/modules/grid_search.html
+
+[Ridge]:https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Ridge.html#sklearn.linear_model.Ridge
+[1]:https://medium.com/distributed-computing-with-ray/gridsearchcv-2-0-new-and-improved-ee56644cbabf
+
+> Cutting edge hyperparameter tuning techniques (bayesian optimization, early
+> stopping, distributed execution) can provide significant speedups over grid
+> search and random search.
+>
+> However, the machine learning ecosystem is missing a solution that provides
+> users with the ability to leverage these new algorithms while allowing users
+> to stay within the Scikit-Learn API. In this blog post, we introduce
+> tune-sklearn [Ray's tuning library] to bridge this gap. Tune-sklearn is a
+> drop-in replacement for Scikit-Learn’s model selection module with
+> state-of-the-art optimization features.
+>
+> —[GridSearchCV 2.0 — New and Improved][1]
+
+This claim is inaccurate: Dask-ML has provided access to "cutting edge
+hyperparameter tuning techniques" with a Scikit-learn compatible API for over a
+year. To provide correct this inaccuracy, let's start by looking at the
+features of Ray's tune-sklearn:
+
+[dml-hod]:https://ml.dask.org/hyper-parameter-search.html
+
+> Here’s what [Ray's] tune-sklearn has to offer:
+> 1. **Consistency with Scikit-Learn API** ...
+> 2. **Modern hyperparameter tuning techniques** ...
+> 3. **Framework support** ...
+> 4. **Scale up** ...
+>
+> [Ray's] Tune-sklearn is also **fast**.
+
+Dask-ML's model selection module has every one of the features. Let's look at
+each feature and see what Dask-ML and Ray have to offer:
+
+> 1\. **Consistency with Scikit-Learn API**: tune-sklearn is a drop-in
+> replacement for GridSearchCV and RandomizedSearchCV, so you only need to
+> change less than 5 lines in a standard Scikit-Learn script to use the API.
+
+*Dask-ML is also consistent with the Scikit-learn API.*
+
+Here's how to use Dask-ML's hyperparameter optimization:
+
+``` python
+from dask_ml.model_selection import HyperbandSearchCV
+from dask.distributed import Client
+from scipy.stats import loguniform
+from sklearn.datasets import make_classification
+from sklearn.linear_model import SGDClassifier
+
+if __name__ == "__main__":
```

Comment:
Nitpick, but I think it's easier to read this without the `if __name__ == "__main__":` bit.

Reply:
Yeah, that's a good idea. It won't run without the `if __name__ == "__main__":` block but it warns otherwise.